### DISCUSSION TITLE
Password wrong even when using urllib.parse.quote_plus, but using pyodbc directly works

### DISCUSSION THREAD

**Discussion by KyllVLP** [sqlalchemy/sqlalchemy#8951] [Usage Questions]
I'm running into a problem when trying to connect to a SQL server.  Creating a normal connect string didn't work to establish a connection, so I went for the other pyodbc route. I use a string that works when I use pyodbc directly. It works for establishing a connection. However, I get the following error:
```
sqlalchemy.exc.InterfaceError: (pyodbc.InterfaceError) 
('28000', "[28000] [Microsoft][ODBC Driver 18 for SQL Server][SQL Server]Login failed for user 'XXXX'. (18456) (SQLDriverConnect); 
[28000] [Microsoft][ODBC Driver 18 for SQL Server][SQL Server]Login failed for user 'XXXX'. (18456)")
```
The code I use:

```
params = urllib.parse.quote_plus('DRIVER={ODBC Driver 18 for SQL Server};SERVER='+sqlServer+';DATABASE='+Database+';UID='+sql_user+';PWD='+sql_password+';TrustServerCertificate=Yes')
print(params)
engine = create_engine('mssql+pyodbc:///?odbc_connect=%s' % params) 
```

If I use the same string for pyodbc directly, the connection works correctly. 
Am I doing something wrong or is this a bug?

**Comment by zzzeek** [comment]
hi -

can you try [programmatic URL construction](https://docs.sqlalchemy.org/en/14/core/engines.html#creating-urls-programmatically) ?  that will eliminate any escaping issues as the problem.  "query" can be passed as a dictionary.
